src/Sentry.Maui/Internal/ScreenshotAttachment.cs

```diff
@@ -0,0 +1,39 @@
+namespace Sentry.Maui.Internal;
+
+internal class ScreenshotAttachment : Attachment
+{
+    public ScreenshotAttachment()
+        : this(
+            AttachmentType.Default,
+            new ScreenshotAttachmentContent(),
+            "screenshot.jpg",
+            "image/jpeg")
+    {
+    }
+
+    private ScreenshotAttachment(
+        AttachmentType type,
+        IAttachmentContent content,
+        string fileName,
+        string? contentType)
+        : base(type, content, fileName, contentType)
+    {
+    }
+}
+
+internal class ScreenshotAttachmentContent : IAttachmentContent
+{
+    public Stream GetStream()
+    {
+        if (Screenshot.Default.IsCaptureSupported)
+        {
+            var screen = Screenshot.Default.CaptureAsync().ConfigureAwait(false).GetAwaiter().GetResult();
```

Comment:
Off there's no way to make this `async` instead? The transport is `async` but I imagine we'd want the screenshot done at the time of the error and not later on in the transport once "the relevant point in time is gone".

Q:
1. Is there a risk of deadlocking here?
2. Does this need to run on the UI thread? What happens if it's called on a backgorund thread (happens if the error is captured on a background thread)

Finally, this is actually only going to run when we call `GetStream` which is indeed on the background worker, could you confirm?
We'd want this to run synchronously when `Capture` is called into the SDK to get the screen at that moment.

Reply:
Just tried it and it's deadlocking on UWP. Shouldn't be an issue on other platforms because on other platforms it runs synchronously and returns `Task.FromResult`

Reply:
Could you please add a comment block with pointers? I think it'll help put ppl's mind at ease: 

```
// This actually runs synchronously (returning Task.FromResult) on the following platforms:
// Android: https://github.com/dotnet/android/.../Bla.cs
// iOS/macOS: https://github.com/dotnet/macios/.../Bla.cs

```

What's the strategy on UWP? Do we skip with `#ifdef` ?

Reply:
Add comments and skip on Windows.